scipy/optimize/_differentialevolution.py

```diff
@@ -116,12 +116,32 @@ def differential_evolution(func, bounds, args=(), strategy='best1bin',
         Specify `seed` for repeatable minimizations.
     disp : bool, optional
         Prints the evaluated `func` at every iteration.
-    callback : callable, `callback(xk, convergence=val)`, optional
-        A function to follow the progress of the minimization. ``xk`` is
-        the best solution found so far. ``val`` represents the fractional
-        value of the population convergence.  When ``val`` is greater than one
-        the function halts. If callback returns `True`, then the minimization
-        is halted (any polishing is still carried out).
+    callback : callable, optional
+        A callable called after each iteration. Has the signature:
+
+            ``callback(intermediate_result: OptimizeResult)``
+
+        where ``intermediate_result`` is a keyword parameter containing an
+        `OptimizeResult` with attributes ``x`` and ``fun``, the best solution
+        found so far and the objective function. Note that the name
+        of the parameter must be ``intermediate_result`` for the callback
+        to be passed an `OptimizeResult`.
+
+        The callback also supports a signature like:
+
+            ``callback(x, convergence: float =val)``
+
+        ``val`` represents the fractional value of the population convergence.
+        When ``val`` is greater than one the function halts.
```

Comment:
if the solve has converged I would expect that val be greater than 1. I prefer using words rather than numbers in contexts like this.

Reply:
> I prefer using words rather than numbers in contexts like this.

The trouble with "one" is that it can temporarily be misunderstood to mean, e.g., one of the population. That doesn't make sense in context, so one can come to the correct understanding. But when there is more than one meaning of a word, it is just a little harder to parse. This is why I suggested "unity".